### PR TITLE
[3/4] Improve: Small Cleanup / Tech Debt Reduction Around NexusWebApi / V2 Types.

### PR DESCRIPTION
**Pull request by Sewer56**
This is 3rd PR in chain. Continues from:

- #2092

Should be merged after that PR, this will reduce the diff from main to only changed parts.

Next in chain (final PR):

- #2113

Review final PR if you want to review all parts at once, else review them by parts sequentially.

----------

Also contributes a bit towards:

- https://github.com/Nexus-Mods/NexusMods.App/issues/2091#issue-2549044273

----------

# Summary

This PR moves the types from 
- #2092

Into the mainline `NexusWebApi` abstractions project.
I refrained from doing that in #2092 because it would make PR reviewing very annoying.

In addition some work is done to help migration towards the V2 API.

## Changelog

- `ModUpdates` project now uses types in `NexusWebApi` abstractions.
- Types which are confirmed to exist in V2 (GraphQL) API are placed in a `V2` folder to prevent accidental future deletion when getting rid of REST APIs.
- The structs `ModId`, `FileId`, `GameId` now use the correct field size (`u32`)
   - They are *not* `u64` like in our previous definition. This has been verified with both backend and GraphQL documentation.
- Implemented `UidForFile` and `UidForMod`.
    - These allow us to split up UID composite keys returned from the website into their specific components, (`GameId`+ `ModId`) or (`GameId`+`FileId`).
    - Methods to decode strings in V2 API results into `Uid` variants are provided.

### CODE REVIEW

**Review by halgari — CHANGES_REQUESTED** [review]
Didn't mean to approve it above

File: tests/Networking/NexusMods.Networking.ModUpdates.Tests/Tests.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+namespace NexusMods.Networking.ModUpdates.Tests;
+
+public class Tests
+{
+    [Fact]
+    public void Test1() { }
```
[the remark this answers is not in the document]
Comment (Sewer56) [reply tests/Networking/NexusMods.Networking.ModUpdates.Tests/Tests.cs:6]: Will fix in previous PR in chain.

File: src/Abstractions/NexusMods.Abstractions.NexusWebApi/Types/V2/Uid/UidForFile.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/// This represents a unique ID of an individual file as stored on Nexus Mods.
+/// 
+/// This is a composite key of <see cref="FileId"/> and <see cref="GameId"/>, where
+/// the upper 4 bytes represent the <see cref="FileId"/> and the lower 4 bytes represent
+/// the <see cref="GameId"/>.
+///
+/// This is consistent with how the Nexus Mods backend produces the UID and is not
+/// expected to change.
+/// </summary>
+[StructLayout(LayoutKind.Sequential, Pack = 1)]
```
[the remark this answers is not in the document]
Comment (Sewer56) [reply src/Abstractions/NexusMods.Abstractions.NexusWebApi/Types/V2/Uid/UidForFile.cs:15]: Consider it done.
I'll make a number of sample tests based on API responses.

[the remark this answers is not in the document]
Comment (Sewer56) [reply src/Abstractions/NexusMods.Abstractions.NexusWebApi/Types/V2/Uid/UidForFile.cs:15]: Done in e8e09e3213447437f232977646393573ca0cdf77

File: tests/Networking/NexusMods.Networking.ModUpdates.Tests/Tests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sewer56) [reply tests/Networking/NexusMods.Networking.ModUpdates.Tests/Tests.cs]: Done in e4349be69b8ef1fee3a2de7612df9ed01b4f8dd6

**Pull request merged**